demos/browser/app/meetingV2/meetingV2.ts

```diff
@@ -2325,6 +2364,48 @@ export class DemoMeetingApp
     }
   }
 
+  private async populateVideoPreviewFilterInputList(): Promise<void> {
+    const genericName = 'Filter';
+    let filters: VideoFilterName[] = ['None'];
+    
+    if (
+      this.defaultBrowserBehaviour.supportsCanvasCapturedStreamPlayback() &&
+      this.enableUnifiedPlanForChromiumBasedBrowsers
+    ) {
+      filters = filters.concat(VIDEO_FILTERS);
+      if (platformCanSupportBodyPixWithoutDegradation()) {
+        if (!this.loadingBodyPixDependencyPromise) {
+          this.loadingBodyPixDependencyPromise = loadBodyPixDependency(this.loadingBodyPixDependencyTimeoutMs);
+        }
+        // do not use `await` to avoid blocking page loading
+        this.loadingBodyPixDependencyPromise.then(() => {
+          filters.push('Segmentation');
+          this.populateFilterList(
+            'video-input-filter',
+            genericName,
+            filters
+          );
+        }).catch(err => {
+          this.log('Could not load BodyPix dependency', err);
+        });
+      }
+    }
+
+    this.populateFilterList(
+      'video-input-filter',
+      genericName,
+      filters
+    );
+  }
+
+  private async discardVideoFilterStream(): Promise<void>  {
+    this.log('Clearing filter variables and discarding the video stream');
+    this.chosenVideoFilter = 'None';
+    this.selectedVideoFilterItem = 'None';
+    await this.audioVideo.chooseVideoInputDevice(null);
+    this.chosenVideoTransformDevice?.stop();
+  }
```

Comment:
I tihnk the man point of this is not to discard the video stream but rather to stop the video processor